### PR TITLE
Changes to disable DDU trigger plots in DT DQM Online Trigger checks …

### PR DESCRIPTION
**Pull request by jfernan2**
…and disable check on RPC paths for TriggerEfficiency studies in DT DQM Offline. Port to 90X of PR 16679
Detailed description:
- DDU plots removed, only TwinMux (TM) now
- Plots on DT/03-LocalTrigger-TM which were empty up to now, should now be filled: TM_TrigEff*Denum, TM_TrigEff*Num and TM_TrigEff*

### CODE REVIEW

**Comment by dmitrijus** [comment]
+1

**Comment by dmitrijus** [comment]
+1

**Comment by dmitrijus** [comment]
+1

File: DQM/DTMonitorModule/python/dtDQMOfflineSources_cff.py [hunk 1]
```diff
@@ -14,6 +14,21 @@
                          subSystemFolder = cms.untracked.string('DT')
                          )
 
+# DT digitization and reconstruction
+# Switched to TwinMux
+from EventFilter.L1TXRawToDigi.twinMuxStage2Digis_cfi import *
```
Comment (davidlange6) [inline DQM/DTMonitorModule/python/dtDQMOfflineSources_cff.py]: hi @jfernan2 -all of the changes in this file need to be moved into the EventFilter package itself. (and the unpackers go into the raw2digi step not the dqm one.

Comment (jfernan2) [inline DQM/DTMonitorModule/python/dtDQMOfflineSources_cff.py]: hi @davidlange6 
Done. Should I do the same to the equivalent 80X and 81X PR too?
Thanks

**Comment by cvuosalo** [comment]
@cmsbuild please test

**Comment by cvuosalo** [comment]
@jfernan2: Please add to the PR description which DQM plots are being changed and how they should change. The Jenkins results show DQM plot changes that are a bit difficult to interpret.

Here are a few from workflow 	25202.0_TTbar_13 that show entries disappearing from one plot, entries being added to another, and efficiency dropping in yet another. It's hard to tell if this PR is working as intended.

eff
denum
phi

**Comment by jfernan2** [comment]
@cvuosalo thanks for checking this!
I have added the description of the changes.
I believe the plots are fine, up to now were empty due to RPC filter, now they are filled if stats allow, for example in ZMuMu sample:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_9_0_X_2016-12-07-0000+16739/17415/1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15/DT_03-LocalTrigger-TM_Task.html

The TTbar sample should not have many muon statistics, it could be fluctuations, however I'm not able to find the plots you point out for a deeper inspection in the Jenkins comparison. I do see this with entries now where there were no entries before:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_9_0_X_2016-12-07-0000+16739/17415/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25/DT_03-LocalTrigger-TM_Task.html

tanks

**Comment by cvuosalo** [comment]
@jfernan2: You can find the DQM plots here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_9_0_X_2016-12-07-0000+16739/17415/alternative-comparisons/
The workflows that show significant differences are: 4.22, 9.0, 25, 136.731, 140.53, 1000, 1330, 10021, 10024, 20034, 21234, 23234, 25202, and 50202.

You say TM_TrigEff plots are now filled. I see some getting filled, while others become empty. Here are examples of plots getting emptied from workflow 4.22_RunCosmics2011A (left is baseline, right is PR):

moremisseff
misseff

**Comment by cvuosalo** [comment]
If there is no progress on this PR today (Friday), it will probably miss 900pre2.

**Comment by jfernan2** [comment]
Thanks, @cvuosalo but I'm working on demonstrating that the problem in the Cosmics workflow  is due to the RAW version of the data (2011), where the DDU was used. Now this PR is disabling the DDU due to the introduction of the TwinMux boards, that's why the baseline has non-empty histograms. I would suggest moving the Cosmic workflow in Jenkins test to 2016 Cosmic data for RAW data compatibility with phase I upgrades. 

In the other workflows I believe the differences are due to statistics though.
However I will not have the proofs until next week. Thanks.

**Comment by jfernan2** [comment]
@cvuosalo As I pointed out on Friday, I have run myself the 2011 Cosmics workflow over a large statistics sample (100k events) with the PR and produced the DQM file where you can see that there are no empty histograms.
- Root file: /afs/cern.ch/user/j/jfernan/work/public/DTDQM_Cosmics2011/DQM_V0001_R000160960__Global__CMSSW_X_Y_Z__RECO.root
- Uploaded to my DQM GUI: http://fanae43.geol.uniovi.es:8080/dqm/offline/start?runnr=160960;dataset=/Cosmics/Run2011A/DQMIO;sampletype=offline_data;filter=all;referencepos=overlay;referenceshow=customise;referencenorm=True;referenceobj1=refobj;referenceobj2=none;referenceobj3=none;referenceobj4=none;search=;striptype=object;stripruns=;stripaxis=run;stripomit=none;workspace=DT;size=M;root=DT/03-LocalTrigger-TM;focus=;zoom=no;

As I mentioned, the reason why the baseline has non-empty histograms while the PR does, is double: 2011 RAW data format included DDU (and RPC hits in trigger from it), which made populate the histograms with the DDU and RPC filter enabled, even with stats as low as 10 events (as the Jenkins test use).

In 2016, with the introduction of the TwinMux boards, all the DT DQM Offline plots in folder  DT / 03-LocalTrigger-TM are empty due tow these two facts above. You can check in the DQM Offline Gui in any decent Cosmic or Data run, e.g.:
https://goo.gl/fG722v

I'm pretty sure the PR is doing well, but I'm willing to perform any other test or workflow you consider the differences are salient with 10 event statitics, as Jenkins test is using. Nevertheless, for the future I'd suggest moving the Cosmics Jenkin test to a recent 2016 dataset as reference to avoid this kind of problems.

Thanks

**Comment by slava77** [comment]
can we use configuration changes only to preserve pre-2016 behavior?

If so, then this PR should use era feature and keep pre-2016
functionality in place.

**Comment by jfernan2** [comment]
@slava77 Thanks. But I'm not sure if I understand your comment/request. 
Let me put this way:
The whole 2016 data taking have those DQM plots empty due to the use of new Trigger Primitives in the DT data format. We realized now the source of the problem so, only in case a pre-2016 data format is used as input (like the 2011 Cosmics used in the Jenkins test) the plots have entries in the baseline.

This change can go to 90X only if it eases the merging, though we would like to have it in 8XY too to take advantage of any data reprocessing, but I guess there will be no further data reprocessings anyway in a short-term time scale...
Thanks

**Comment by slava77** [comment]
On 12/12/16 6:27 AM, jfernan2 wrote:
> @slava77 <https://github.com/slava77> Thanks. But I'm not sure if I
> understand your comment/request.

If we can support both pre-2016 and 2016+  by just adjusting configuration,
then I think we should.
So that the plost are filled in a meaningful way in all scenarios.

Our configuration language allows this support using eras.

If code that supports pre-2016 is incompatible with that for 2016+,
then DQM decides if it's time to abandon support for pre-2016,
then we do what this PR has now (broken or descoped monitoring of
pre-2016 data)
(chances are that someone may still need to check some old data)

**Comment by jfernan2** [comment]
@slava77 Thanks, now I understand what you mean. I can add an extra configuration parameter so that the RPC filter can be enabled/disabled via python. However, I don't now how to set that this be used depending on the input Era, can you point to an example?

On the other hand, these DQM plots are used by DT expert shifter, who knows exactly this incompatibility, so I'm not sure if it may be needed or not.
Thanks

**Comment by slava77** [comment]
On 12/13/16 12:06 AM, jfernan2 wrote:
> @slava77 <https://github.com/slava77> Thanks, now I understand what you
> mean. I can add an extra configuration parameter so that the RPC filter
> can be enabled/disabled via python. However, I don't now how to set that
> this be used depending on the input Era, can you point to an example?
> 
> On the other hand, these DQM plots are used by DT expert shifter, who
> knows exactly this incompatibility, so I'm not sure if it may be needed
> or not.
> Thanks


https://github.com/cms-sw/cmssw/blob/53327033fb356b2a704464439ce79fa47a9d1678/CalibMuon/CSCCalibration/python/CSCChannelMapper_cfi.py#L15
has an example of using eras
for things installed in 2016 you can use an existing run2_25ns_specific
modifier.

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by cvuosalo** [comment]
@jfernan2: What is the purpose of the latest era change to this PR? I checked the new Jenkins test results, and they seem the same as I reported before the latest change (https://github.com/cms-sw/cmssw/pull/16739#issuecomment-265584289).

**Comment by cvuosalo** [comment]
+1

For #16739 38e35b4fede974885c318bde37fc0f2bc7c2d04a

Changing some DT and RPC DQM trigger plots.

The Reco code change is satisfactory. Jenkins tests against baseline CMSSW_9_0_X_2016-12-14-1100 show no Reco problems, but there still might be issues of concern to DQM.
@dmitrijus: It appears that even the latest version of this PR is failing to preserve pre-2016 behavior in some of the DQM plots. Reco will defer to DQM as to how to best proceed from here.

**Comment by jfernan2** [comment]
Please hold on, I'm still working on a solution for the DQM differences, but not much time available lately. Hope to arrive to commit by today

**Comment by jfernan2** [comment]
Now it should be fine, thanks for your patience

**Comment by cvuosalo** [comment]
@cmsbuild please test

**Comment by cvuosalo** [comment]
+1

For #16739 8f292b6e50f37996c46ae7f316beb453e7a94495

Second approval, after DQM code fixes. Jenkins test results show only TM_TrigEff DQM plots getting filled in as desired, with no plots being erroneously emptied as seen previously before the code fix. Note that DQM plots for workflows 136.731 and 1000.0 are completely wrong, but that is a long-standing Jenkins problem not related to this PR.

File: DQM/DTMonitorModule/python/dtDQMOfflineSources_Cosmics_cff.py [hunk 2]
```diff
@@ -42,11 +46,12 @@
...
 dtSourcesCosmics = cms.Sequence(dtDataIntegrityUnpacker  +
-                                DTDataIntegrityTask +
-                                dtDCSByLumiMonitor + 
-                                dtRunConditionVar + 
-                                dtSegmentAnalysisMonitor +
-                                dtResolutionAnalysisMonitor +
-                                dtEfficiencyMonitor +
-                                dtTriggerEfficiencyMonitor +
-                                dqmInfoDT)
+			 twinMuxStage2Digis +
```
Comment (dmitrijus) [inline DQM/DTMonitorModule/python/dtDQMOfflineSources_Cosmics_cff.py]: tabs, please switch to spaces!

File: DQM/DTMonitorModule/python/dtDQMOfflineSources_cff.py [hunk 3]
```diff
@@ -42,6 +46,7 @@
 from DQM.DTMonitorModule.dtTriggerEfficiencyTask_cfi import *
 
 dtSources = cms.Sequence(dtDataIntegrityUnpacker  +
+			 twinMuxStage2Digis +
```
Comment (dmitrijus) [inline DQM/DTMonitorModule/python/dtDQMOfflineSources_cff.py]: tabs again!

**Review by dmitrijus — CHANGES_REQUESTED** [review]
Please fix the white space issues in python configuration files. Also, https://twiki.cern.ch/twiki/bin/viewauth/CMS/DQMGithubGuide

**Comment by slava77** [comment]
@cmsbuild please test

File: DQM/DTMonitorModule/python/dtDQMOfflineSources_HI_cff.py [hunk 4]
```diff
@@ -35,13 +39,13 @@
...
     )
 )
-
 from DQM.DTMonitorModule.dtDataIntegrityTask_cfi import *
 DTDataIntegrityTask.processingMode = "Offline"
 
 from DQM.DTMonitorModule.dtTriggerEfficiencyTask_cfi import *
 
 dtSources = cms.Sequence(dtDataIntegrityUnpacker  +
+			 twinMuxStage2Digis  +
```
Comment (dmitrijus) [inline DQM/DTMonitorModule/python/dtDQMOfflineSources_HI_cff.py]: Another tab!

**Review by dmitrijus — CHANGES_REQUESTED** [review]
Please just configure your editor not to emit tabs. Ever.
This as frustrating to me as it is to you.

**Comment by jfernan2** [comment]
Done. My apologies

**Comment by slava77** [comment]
@cmsbuild please test

File: DQM/DTMonitorModule/python/dtDQMOfflineSources_HI_cff.py [hunk 5]
```diff
@@ -45,7 +45,7 @@
 from DQM.DTMonitorModule.dtTriggerEfficiencyTask_cfi import *
 
 dtSources = cms.Sequence(dtDataIntegrityUnpacker  +
-			 twinMuxStage2Digis  +
+	                 twinMuxStage2Digis  +
```
Comment (dmitrijus) [inline DQM/DTMonitorModule/python/dtDQMOfflineSources_HI_cff.py]: Still a tab left. Now it is even worse, tab + space + more space?
(also please check the other files...)

**Review by dmitrijus — CHANGES_REQUESTED** [review]

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/16739	ff549f2
- only DQM/DT\* files were modified since the last signoff by reco https://github.com/cms-sw/cmssw/compare/38e35b4fede974885c318bde37fc0f2bc7c2d04a...ff549f2b7a24ca9a1a17a2a317852d731769b2d9
    - reco products are not changed by this PR, it has DQM changes only
- jenkins tests pass and comparisons with baseline roughly correspond to what was noticed in the last RECO signoff https://github.com/cms-sw/cmssw/pull/16739#issuecomment-268105499

**Comment by dmitrijus** [comment]
+1

File: DQM/DTMonitorModule/python/dtDQMOfflineSources_Cosmics_cff.py [hunk 6]
```diff
@@ -42,11 +46,12 @@
...
 dtSourcesCosmics = cms.Sequence(dtDataIntegrityUnpacker  +
-                                DTDataIntegrityTask +
-                                dtDCSByLumiMonitor + 
-                                dtRunConditionVar + 
-                                dtSegmentAnalysisMonitor +
-                                dtResolutionAnalysisMonitor +
-                                dtEfficiencyMonitor +
-                                dtTriggerEfficiencyMonitor +
-                                dqmInfoDT)
+                         twinMuxStage2Digis +
```
Comment (davidlange6) [inline DQM/DTMonitorModule/python/dtDQMOfflineSources_Cosmics_cff.py]: Hi @jfernan2 - sorry for not re-reviewing this after your changes - also the producer(s) themselves need to be moved to a path with producers - in this case the L1 sequence I think. Maybe @rekovic  can help.

Comment (jfernan2) [inline DQM/DTMonitorModule/python/dtDQMOfflineSources_Cosmics_cff.py]: Any hint where this should go instead?

Comment (jfernan2) [inline DQM/DTMonitorModule/python/dtDQMOfflineSources_Cosmics_cff.py]: Hi @davidlange6 , @rekovic but, I'm not able to find where should I move the twiMuxStage2Digis in this DQM sequence. Could you please point me to the right package? 

I can trace back to see that DQMOffline/Configuration/python/DQMOffline_cff.py is calling this config file, but I ignore where all the raw2digi is being set for DQM, if so. Thanks

Comment (jfernan2) [inline DQM/DTMonitorModule/python/dtDQMOfflineSources_Cosmics_cff.py]: Hi @davidlange6 , @rekovic , @dmitrijus , could any help me to understand where the unpacker should go instead? 
For DQM online, it is sourced through DQM/DTMonitorModule/python/dt_dqm_sourceclient_common_cff.py in DQM/Integration/python/clients/dt_dqm_sourceclient-live_cfg.py 
But the equivalent for DQM offline seems to be DQM/DTMonitorModule/python/dtDQMOfflineSources_*.py
Any hint is very apreciated
Thank you very much

Comment (dmitrijus) [inline DQM/DTMonitorModule/python/dtDQMOfflineSources_Cosmics_cff.py]: I am not sure what David meant. L1 reco sequence? To me, dtDQMOfflineSources_*.py looks a proper place for it. Ping, @davidlange6 @rekovic

Comment (jfernan2) [inline DQM/DTMonitorModule/python/dtDQMOfflineSources_Cosmics_cff.py]: @dmitrijus Could you anyway trigger the Jenkins tests with the last commit which may fix the situation? I'm not able to run them in my local release in due a strange error not related to this PR: 
"plugin-poisoned-CondFormatsCTPPSReadoutObjectsPlugins.so cannot open shared object file"
Thanks

**Comment by dmitrijus** [comment]
please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/16739	d181328
- DQM-related updates since the last commit, no impact on reco; changes look reasonable (removal of the twinMuxStage2Digis  producer from purely DQM sequences)
- jenkins tests pass; comparisons with baseline look consistent with the differences observed in the past in the DT LocalTrigger monitoring plots

**Comment by dmitrijus** [comment]
+1

**Comment by davidlange6** [comment]
+1
Thanks

**Pull request merged**